pkg/trace/agent/agent_test.go

```diff
@@ -971,32 +972,47 @@ func TestSample(t *testing.T) {
 		return pt
 	}
 	tests := map[string]struct {
-		trace       traceutil.ProcessedTrace
-		wantSampled bool
+		trace               traceutil.ProcessedTrace
+		wantSampled         bool
+		sampleTracerDropped bool
```

Comment:
Another way of writing these tests would be to have to bools: one for whether or not this trace should be sampled *with* the feature flag, and one for whether or not it should be sampled *without* the feature flag. Then in the `t.Run`, you can run your sampling and assertion and check against the first bool, then set the feature flag and do it again, checking against the second bool. That would make sure we cover every case with the feature flag on and off as well (for example, right now there are no tests with UserKeep and the feature flag off)
I think it simplifies the test cases a bit as well.